trino/auth.py

```diff
@@ -150,30 +135,26 @@ def handle_redirect_auth_url(auth_url):
     print(auth_url)
 
 
-class OAuth2Authentication(Authentication):
-
+class _OAuth2TokenBearer(AuthBase):
+    """
+    Custom implementation of Trino Oauth2 based authorization class to get the bearer token
+    """
     MAX_OAUTH_ATTEMPTS = 10
 
     def __init__(self, redirect_auth_url_handler=handle_redirect_auth_url):
         self._redirect_auth_url = redirect_auth_url_handler
-        self._statement_url = ""
-
-    def set_client_session(self, client_session):
-        pass
+        self.token = None
 
-    def set_http_session(self, http_session):
-        self.__authenticate(http_session)
-        return http_session
-
-    def set_statement_url(self, statement_url):
-        self._statement_url = statement_url
-
-    def get_exceptions(self):
-        return ()
+    def __call__(self, r):
+        if self.token is None:
+            self.__authenticate(r.url)
+        r.headers["Authorization"] = "Bearer " + self.token
+        return r
 
-    def __authenticate(self, http_session):
-        http_response = http_session.post(
-            self._statement_url,
+    def __authenticate(self, url):
+        session_to_get_token = requests.Session()
```

Comment:
You're using a different `Session` than the one used by the client. This means that some configuration options will be ignored when requesting a token, for example: `Verify=False`.

Reply:
I need to use a different session otherwise the implementation is going to be simlar to `HTTPDigestAuth,` . Do you think I can just use the http_session properties which are essential for the new session? I just see that one `Verify` property being set?

Reply:
> I need to use a different session otherwise the implementation is going to be simlar to HTTPDigestAuth

I'm not sure I understand. How it's going to be similar and why it shouldn't be?

> Do you think I can just use the http_session properties which are essential for the new session

The problem when using different sessions is that in the future any changes in/additions to the session will have to be done in 2 places: in `client.py` and here.

> I just see that one Verify property being set?

I see `http_session.verify` and `http_session.headers`.